controlplane/modules/route/internal/discovery/neigh/entry.go

```diff
@@ -11,13 +12,22 @@ import (
 type NeighbourEntry struct {
 	// NextHop is the IP address of the next hop.
 	NextHop netip.Addr
-	// LinkAddr is the MAC address of the next hop.
-	LinkAddr net.HardwareAddr
-	// HardwareAddr is the MAC address of the local interface that observed
-	// the neighbour.
-	HardwareAddr net.HardwareAddr
+	HardwareRoute
```

Comment:
Let's leave explicit fields - it's more readable. But you can change their type to [6] byte.

I personally don't honor embedded structs, because there are too much magic. For example this struct now have `String()` method.